connectors/activemq/src/main/java/forklift/producers/ActiveMQProducer.java

```diff
@@ -185,51 +204,48 @@ private Message prepAndValidate(ForkliftMessage message,
     private Message forkliftToJms(ForkliftMessage message) throws ProducerException {
 
         try {
-            Message msg = null;
-            if (message.getJmsMsg() != null ) {
-                try {
-                    msg = message.getJmsMsg();
-                } catch (Exception e) {
-                    throw new ProducerException("Error assigning Forklift JMS message to new MSG", e);
-                }
-            } else if (message.getMsg() != null ) {
-                try {
-                    msg = session.createTextMessage(message.getMsg());
-                } catch (Exception e) {
-                    throw new ProducerException("Error creating new jms TextMessage", e);
-                }
-            }
+            final Message msg = session.createTextMessage(message.getMsg());
+
+            // Set the jms correlation id.
+            if (message.getId() != null && !message.getId().trim().equals(""))
+                msg.setJMSCorrelationID(message.getId());
+
             setMessageHeaders(msg, message.getHeaders());
             setMessageProperties(msg, message.getProperties());
             return msg;
         } catch (Exception e) {
-            throw new ProducerException("Error creating JMS Message, Forklift message was null", e);
+            throw new ProducerException("Error creating JMS Message, or Forklift message was null", e);
         }
     }
 
     private void setMessageHeaders(Message msg, Map<Header, Object> headers) throws ProducerException {
         if (msg != null && headers != null) {
             headers.entrySet().stream().filter(entry -> entry.getValue() != null)
-                              .forEach(entry -> {
-                                if(ActiveMQHeaders.getFunctions().get(entry.getKey()).get((ActiveMQMessage)msg) == null) {
-                                   ActiveMQHeaders.getFunctions().get(entry.getKey()).set((ActiveMQMessage)msg,entry.getValue());
-                                }
-                            });
+                .forEach(entry -> {
```

Comment:
Do we want to also factor in stored properties here, like in `setMessageProperties`?